src/app/space/create/deployments/resource-usage/deployments-resource-usage.component.html

```diff
@@ -11,9 +11,18 @@ <h3>
             </b>
           </h3>
         </div>
-        <div [collapse]="resourcesCollapsed">
-          <div class="col-xs-12 col-sm-4 col-md-3" *ngFor="let environment of environments | async">
-            <resource-card [spaceId]="spaceId | async" [environment]="environment"></resource-card>
+        <div>
+          <div [collapse]="resourcesCollapsed">
+            <div *ngIf="environments | async as environmentsList; else loading">
+              <div class="col-xs-12 col-sm-4 col-md-3" *ngFor="let environment of environmentsList">
+                <resource-card [spaceId]="spaceId | async" [environment]="environment"></resource-card>
+              </div>
+            </div>
+            <ng-template #loading>
+              <div class="col-xs-12 col-sm-4 col-md-3" *ngFor="let num of [0, 1]">
+                <loading-resource-card></loading-resource-card>
```

Comment:
Rather than add a `loading-resource-card` template and having it be replaced by the `resource-card` when loaded, I would prefer the `resource-card` to have the initial loading state, where the side text displays 'loading' and the percent is 0. When data comes in, it will update appropriately. The loading state as a separate template should not be necessary.

Reply:
+1

Reply:
@jiekang While I agree this would be nicer, I don't quite understand how it can be done. The resource card takes two inputs - an environment and a string - and in the time where the environments observable is waiting for a value, at least one of these won't be available. With that as the case, resource-card's ngOnInit will fail. 

I can see that the change would be to wait for the environment to become available, then run ngOnInit, but is that possible in js ? Moreover, in deployments-resource-usage.component.html, I would still need to hard code two resource cards to the loading template to fit our design spec. Wouldn't these just be replaced anyways when the ngIf recognised that the observable had received a value ?